refactor(ui):  artifacts table Fixes #8306

Unperceivable opened the pull request:
<!-- markdownlint-disable MD041 -- this is rendered within existing HTML, so allow starting without an H1 -->

<!--

### Before you open your PR

- Run `make pre-commit -B` to fix codegen and lint problems (build will fail).
- [Signed-off your commits](https://github.com/apps/dco/) (otherwise the DCO check will fail).
- Used [a conventional commit message](https://www.conventionalcommits.org/en/v1.0.0/).

### When you open your PR

- PR title format should also conform to [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).
- "Fixes #" is in both the PR title (for release notes) and this description (to automatically link and close the issue).
- Create the PR as draft.
- Once builds are green, mark your PR "Ready for review".

When changes are requested, please address them and then dismiss the review to get it reviewed again.

-->

<!-- Does this PR fix an issue -->

Fixes #8306

### Motivation

From original [issue](https://github.com/argoproj/argo-workflows/issues/8306)
> It takes time to read and understand artifact info now, especially on first columns whose values that may merge with each other.
(https://github.com/argoproj/argo-workflows/issues/8306)
![previous implementation](https://user-images.githubusercontent.com/48029150/161403294-4db609a5-241e-48a9-875d-3d15ed60d8f6.png)

<!-- TODO: Say why you made your changes. -->

### Modifications

Refactored workflow-details artifacts white-box to include table columns and expand a cell with text too long for it's cell on hover.
![image](https://github.com/user-attachments/assets/126bba71-74ac-4c99-b052-677f64f81398)

Added same functionality to workflow-node-info -> inputs/outputs
![image](https://github.com/user-attachments/assets/3279bac0-d68c-4c1f-a5ae-7576320ef1c3)

### Verification

<!-- TODO: Say how you tested your changes. -->

<!--
### Beyond this PR

Thank you for submitting this! Have you ever thought of becoming a Reviewer or Approver on the project?

Argo Workflows is seeking more community involvement and ultimately more [Reviewers and Approvers](https://github.com/argoproj/argoproj/blob/main/community/membership.md) to help keep it viable.
See [Sustainability Effort](https://github.com/argoproj/argo-workflows/blob/main/community/sustainability_effort.md) for more information.

-->

tczhao commented on ui/src/workflows/components/workflow-artifacts.tsx:
```diff
@@ -40,21 +40,34 @@ export function WorkflowArtifacts(props: Props) {
         <div className='white-box'>
             <div className='white-box__details'>
+                <div className='row header'>
+                    <div className='columns download'>Download</div>
```
The download symbol in each row is self-explanatory enough.
we probably don't need this download column, or download column name, what do you think?

tczhao reviewed (changes requested):
The table seems to resize when hovered at the cell, could you fix that

Unperceivable commented:
> The table seems to resize when hovered at the cell, could you fix that  
That was the intended change, from Modifications:  
> expand a cell with text too long for it's cell on hover.  
The alternative suggested in the ticket was a horizontal scroll feature. I figured there would be edge cases where some columns would not appear until you scroll.  
I am fine implementing whatever solution though.

tczhao reviewed (changes requested):
Thank you for the fix, last few things:
1. Could you make sure the table values align with the column name, e.g., left align? People will definitely raise an issue about this later on.
![image](https://github.com/user-attachments/assets/03b88749-e641-4e1f-ab37-0df34030a6b5)
2. we are already on artifact panel, rather than "Artifact Name", using "Name" is probably sufficient enough
![image](https://github.com/user-attachments/assets/24bc90ad-1a55-4d15-a898-0233c40cd0f8)

4. I think we better have the timeformat switch button at the column title instead of individual artifact table value, just like the workflow list page. (we could tackle this in a separate PR, create an issue for this and a separate PR)
![image](https://github.com/user-attachments/assets/fa84d944-d8dd-449b-b62f-05d9892478a2)

Unperceivable commented:
> 1. Could you make sure the table values align with the column name, e.g., left align? People will definitely raise an issue  

Fixed padding issue ef7e1a11b  
> 2. we are already on artifact panel, rather than "Artifact Name", using "Name" is probably sufficient enough

Renamed column. but not in the node overview since there is no label to define artifacts in input/output 2f3b9da87   
> 3. I think we better have the timeformat switch button at the column title instead of individual artifact table value, just 
 like the workflow list page. (we could tackle this in a separate PR, create an issue for this and a separate PR)   

Moved timestamp switch to column fe5110658  
![image](https://github.com/user-attachments/assets/e0a684b2-caed-4d94-bd61-01f71118d32a)  
![image](https://github.com/user-attachments/assets/242796a6-9f44-4b6e-87a7-0f11a9a68b37)

tczhao reviewed (approved):
looks great! Thanks for the contribution and happy new year
